crates/rust-analyzer/src/lsp/to_proto.rs

```diff
@@ -393,16 +395,15 @@ fn completion_item(
             Vec::new()
         };
     let (ref_resolve_data, resolve_data) = if something_to_resolve || !imports.is_empty() {
-        let mut item_index = acc.len();
         let ref_resolve_data = if ref_match.is_some() {
             let ref_resolve_data = lsp_ext::CompletionResolveData {
                 position: tdpp.clone(),
                 imports: Vec::new(),
                 version,
                 trigger_character: completion_trigger_character,
-                completion_item_index: item_index,
+                for_ref: true,
```

Comment:
Out of the same item, we can create two completion items, so keep another property in the resolve data to have the hashing code without any complex conditions but still distinct for both cases.

Reply:
isn't most of the moved fields `Copy` here? I'd expect partial cloning to not be that bad

Reply:
Thank you, indeed, it's the `documentation` and `import_to_add` fields partially moved that I had to clone, this way is definitely better.